### PR TITLE
Introduce scheduling into the runtimeClass API

### LINKED ISSUE #72413: runtime-aware scheduling
With runtimeClass users now tend to have heterogeneous runtimes across clusters, we might want to make the scheduler runtime-aware to determine if node can host a pod requiring a specific runtime/runtimeClass.

This has the benefit to abstract topology knowledge of the cluster to end-users.

cc @bsalamat @tallclair 

/assign
/sig scheduling
/priority important-soon
/kind feature

### PR DESCRIPTION
**Pull request by yastij**
this is a meta-revert kubernetes/kubernetes#78598

**What type of PR is this?**

/kind api-change
/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**: Fixes #72413

**Special notes for your reviewer**:

/assign @thockin @tallclair 
/sig node
/priority important-soon
/kind feature

**Does this PR introduce a user-facing change?**:

```release-note
Introduce Topolgy into the runtimeClass API
```

### CODE REVIEW

File: staging/src/k8s.io/api/testdata/HEAD/node.k8s.io.v1alpha1.RuntimeClass.json [hunk 1]
```diff
@@ -0,0 +1,60 @@
+{
+  "kind": "RuntimeClass",
```
Comment (yastij) [inline staging/src/k8s.io/api/testdata/HEAD/node.k8s.io.v1alpha1.RuntimeClass.json]: results of `UPDATE_COMPATIBILITY_FIXTURE_DATA=true go test ./vendor/k8s.io/api -run //HEAD` to generate fixtures

**Comment by tallclair** [comment]
@yastij why is this on hold?

**Comment by yastij** [comment]
@tallclair - I'd like to get this lgtm/approved but merge it once the implementation is there

File: staging/src/k8s.io/api/testdata/HEAD/node.k8s.io.v1alpha1.RuntimeClass.json (hunk 1 above)
Comment (liggitt) [reply staging/src/k8s.io/api/testdata/HEAD/node.k8s.io.v1alpha1.RuntimeClass.json]: actually, these should have already been generated... fixed separately in master for picking to 1.15 in https://github.com/kubernetes/kubernetes/pull/79029

Comment (yastij) [reply staging/src/k8s.io/api/testdata/HEAD/node.k8s.io.v1alpha1.RuntimeClass.json]: @liggitt - thanks, reverting this and rebasing it on your PR after it merges

**Comment by liggitt** [comment]
see prior approval of API changes in https://github.com/kubernetes/kubernetes/pull/75744#pullrequestreview-244133955

**Comment by tallclair** [comment]
/hold cancel

@yastij can you rebase this, then let's get it merged.

**Comment by yastij** [comment]
@tallclair - ack

**Comment by tallclair** [comment]
Friendly ping. Let me know if you'd like me to take this over.

**Comment by tallclair** [comment]
I'm taking this over for @yastij. Closing in favor of https://github.com/kubernetes/kubernetes/pull/80825

**Pull request closed without merging**